### ISSUE TITLE
Flickering when disposing WebView control

### ISSUE THREAD

**Issue by noxe** [MicrosoftEdge/WebView2Feedback#650]
**Description**
We have a problem when adding / remoing Tabs in an TabControl - during removing the tab / disposing the webview control it sometimes happens that the WebView2 control is flickering - please see attached video - https://www.screencast.com/t/YYYEXbtwUoI

We use Microsoft.WebView2.FixedVersionRuntime.87.0.664.41.x64 and WebView2 release version 1.0.664.37.

https://www.screencast.com/t/YYYEXbtwUoI

In code we simply do:

        private void button1_Click(object sender, EventArgs e)
        {
            var page = new TabPage("WebView2 Test");
            page.Controls.Add(new WebView2() { Dock = DockStyle.Fill });
            this.tabControl1.TabPages.Add(page);
        }

        private void button2_Click(object sender, EventArgs e)
        {
            var page = this.tabControl1.TabPages[0];
            this.tabControl1.TabPages.Remove(page);
            page.Dispose();
        }

**Version**
SDK:  1.0.664.37.
Runtime: 87.0.664.41.x64 fixed version
Framework: WinForms
OS: Win10

**Repro Steps**
See above and video

https://www.screencast.com/t/YYYEXbtwUoI

**Additional context**

**Comment by noxe** [comment]
What we found so far is we can workaround this by adding this:

protected override void OnHandleDestroyed(EventArgs e)
        {
            var controller = typeof(WebView2).GetField("_coreWebView2Controller", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(this) as CoreWebView2Controller;
            controller.IsVisible = false;

            base.OnHandleDestroyed(e);
        }

We set controller.IsVisible false when HandleDestroyed - since the base call does an set_ParentWindow(IntPtr.Zero) which seems to cause the flickering.

**Comment by douglas-jordan** [comment]
In the video are you referring to what looks like the tab jumping out of the parent window?

**Comment by noxe** [comment]
Yes - at second 14/15 and 21/22 you see it - but it is not the tab - it is the browser window/control

**Comment by noxe** [comment]
Is not related to any tab - also happens with a simple form while closing - see here https://www.screencast.com/t/SQv4XN50T

I am on an surface with 200% dpi if this is relevant.

**Comment by douglas-jordan** [comment]
> Yes - at second 14/15 and 21/22 you see it - but it is not the tab - it is the browser window/control

Yes that is what I meant.  This behavior is what happens when the host window is destroyed or removed before the child window  from another process is removed.  The child window in this case is the browser control.

Your "workaround" to hide the window first is the typical way of deleting a tab or child when the host or parent window is being removed. You could also delete the browser control before removing the tab but you may have another side effect of a blank area momentarily while the host window redraws.

I say this is not a bug as this is the behavior of any child window in a separate process.

**Comment by noxe** [comment]
Well - ok - but shouldnt that be handled correctly by the WebView2 control? It happens across different situations - does not matter where i drop the control on? See above - also in a simple scenario when open/close a window?

**Comment by champnic** [comment]
Thanks for the bug report and great videos! I think you've both identified the issue, and I agree this seems like a teardown experience we should be able to improve. I've added a bug to our backlog to track the fix.

**Comment by noxe** [comment]
👌
